### PR TITLE
feat(function): migrate string length functions to new expression framework

### LINKED ISSUE #6776: `bit_length`

### LINKED ISSUE #6777: `octet_length`

### LINKED ISSUE #6778: `char_length`

### LINKED ISSUE #6779: `character_length`

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Fixes #6776 `bit_length`
Fixes #6777 `octet_length`
Fixes #6778 `char_length`
Fixes #6779 `character_length`

### CODE REVIEW

File: common/functions/src/scalars_v2/string.rs [hunk 1]
```diff
@@ -39,5 +40,40 @@ pub fn register(registry: &mut FunctionRegistry) {
             Ok(())
         },
     );
+    registry.register_with_writer_1_arg::<StringType, NumberType<u64>, _, _>(
```
Comment (sundy-li) [inline common/functions/src/scalars_v2/string.rs:43]: `register_with_writer` is slower than `register_1_arg` I think.  Better to use the iterator which avoids too much extra reallocation.

File: common/functions-v2/src/scalars/string.rs [hunk 2]
```diff
@@ -39,5 +40,28 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+        FunctionProperty::default(),
+        |_| None,
+        |val| match std::str::from_utf8(val) {
+            Ok(s) => s.chars().count() as u64,
+            Err(_) => val.len() as u64,
+        },
+    );
     registry.register_aliases("upper", &["ucase"]);
+    registry.register_aliases("octet_length", &["length"]);
+    registry.register_aliases("char_length", &["character_length"]);
```
Comment (andylokandy) [inline common/functions-v2/src/scalars/string.rs:66]: Maybe it's better to move aliases to near the function definition. What do you think?

**Review by andylokandy — APPROVED** [review]

**Comment by andylokandy** [comment]
@mergify update

**Pull request merged**